dncil/cil/body/__init__.py

```diff
@@ -265,11 +261,14 @@ def parse_basic_blocks(self):
                 # no fallthrough
                 continue
 
-            # connect fallthrough
+            # connect fallthrough:
+            # if a basic block has a fallthrough successor, it will be the next basic block in our list
+            # since instructions, leaders, and basic blocks are processed sequentially
             try:
                 bb_next: BasicBlock = self.basic_blocks[idx + 1]
-                bb.succs.append(bb_next)
-                bb_next.preds.append(bb)
             except IndexError:
                 # end of method
                 continue
+            else:
+                bb.succs.append(bb_next)
+                bb_next.preds.append(bb)
```

Comment:
Can you provide context for this change? What is it addressing?

Reply:
This change was made in accordance to:
https://github.com/mandiant/capa/pull/1326#discussion_r1121662676